src/main/java/com/notably/view/blockcontent/BlockContentEditView.java

```diff
@@ -1,63 +1,253 @@
+public class BlockContentEditView extends ViewPart<Stage> {
 
+    private static final Logger logger = LogsCenter.getLogger(BlockContentEditView.class);
+    private static final String FXML = "blockcontent/BlockContentEditView.fxml";
+    private final Stage stage;
+    private final Window parentStage;
+    private final Logic logic;
     private final Model model;
 
     @FXML
     private TextArea blockContentTextArea;
 
-    public BlockContentEditView(Model model) {
-        super(FXML);
+    public BlockContentEditView(Logic logic, Model model) {
+        this(new Stage(), logic, model);
+    }
 
-        Objects.requireNonNull(model);
+    private BlockContentEditView(Stage root, Logic logic, Model model) {
+        super(FXML, root);
+
+        requireAllNonNull(logic, model);
+
+        this.stage = root;
+        this.logic = logic;
         this.model = model;
+        this.parentStage = Stage.getWindows().stream().filter(Window::isShowing).findFirst().get();
 
-        setup();
+        setStageStyle();
+        setChangeListeners();
+        setInitialData();
     }
 
     /**
-     * Sets up the view's initial data and wire up all required change listeners.
+     * Customizes the appearance and attributes of the Block Edit Modal
      */
-    private void setup() {
-        // Initialize text area content
-        setText(model);
+    private void setStageStyle() {
+        stage.initStyle(StageStyle.TRANSPARENT);
+        stage.initOwner(parentStage);
+        stage.initModality(Modality.WINDOW_MODAL);
+    }
 
-        // Setup change listeners
+    /**
+     * Sets listeners that update the content in the Block Edit modal to that of the
+     * currently opened note, and toggle the visibility of the edit modal.
+     */
+    private void setChangeListeners() {
         model.currentlyOpenPathProperty().addListener(observable -> setText(model));
         model.getBlockTree().getRootBlock().getTreeItem()
                 .addEventHandler(TreeItem.treeNotificationEvent(), event -> setText(model));
+        model.blockEditableProperty().addListener(observable -> {
+            if (model.isBlockEditable()) {
+                handleEdit();
+                model.setBlockEditable(false);
+            }
+        });
+
+        setKeyboardListeners();
+        setInitialDimensions();
+        setStageDimensionListeners();
     }
 
     /**
-     * Sets the {@link TextArea}'s content of with the currently open block's Markdown body.
+     * Triggers changes in the visibility of the Edit modal when certain Keystrokes are registered.
+     * Primarily, allows for the user to exit the Block Edit modal using the "ESC" keyboard shortcut.
+     */
+    private void setKeyboardListeners() {
+        stage.addEventHandler(KeyEvent.KEY_RELEASED, (KeyEvent event) -> {
+            if (KeyCode.ESCAPE == event.getCode()) {
+                handleClose();
+            }
+        });
+    }
```

Comment:
Do you think we should add another shortcut that is more relevant to the general user? Something like `ctrl/cmd + s`

Reply:
Good Suggestion! `ctrl/cmd + s` would be more intuitive. I will set it up such that the same behavior on pressing `ESC` happens with `ctrl/cmd + s`.